### ISSUE TITLE
enhancement proposal: refactor elif statements for partitioners in auto.py 

### ISSUE THREAD

**Issue by Coniferish** [Unstructured-IO/unstructured#1716]
There's a lot of redundancy in auto.py when going through the `elif` statements to check the `filetype` and send the document to the correct partitioner. This proposes collecting all the options users can define into a single object that can be unpacked by each partitioner so partition calls can look something like this instead of having to explicitly pass all the arguments (but still explicitly pass all data source parameters):

```
if filetype == FileType.PDF:
        elements = partition_pdf(
            filename=filename,
            file=file,
            url=url,
            **kwargs,
        )
    if filetype == FileType.HTML:
        elements = partition_html
            filename=filename,
            file=file,
            url=url,
            **kwargs,
        )
```

We could create a `PartitionerOptions(TypedDict)` type that specifies the names and types of all the keyword-value pairs that can appear as options. This could help prevent typos and such and provides general type-safety.

**Comment by Coniferish** [comment]
@scanny Do you have any additional thoughts on what this could/should look like?

**Comment by newelh** [comment]
I sketched some ideas out in a [branch here](https://github.com/Unstructured-IO/unstructured/blob/newelh/metadata-refactor/unstructured/partition/partition.py). It's far from complete, and probably too much of a refactor but some ideas along the same line of thinking

**Comment by Coniferish** [comment]
Oh nice. Is there an open issue that branch is being developed for? I admittedly didn't look for any open Issues related to this, but did just find this one: 
https://github.com/Unstructured-IO/unstructured/issues/601

**Comment by newelh** [comment]
No, there wasn't a specific issue, mostly personal musings over a weekend. It's mildly related to #1521. e.g. reducing the amount of kwargs we're passing around, and long-term simplifying the user experience.

**Comment by scanny** [comment]
Fixed by #3806.
